### DISCUSSION TITLE
Monitoring Heartbeat with uWebSockets.js and socket.io Integration and Updating Device Status

### DISCUSSION THREAD

**Discussion by iakuf** [uNetworking/uWebSockets.js#1129] [Q&A] [answered]
I am using this approach to integrate socket.io with uWebSockets.js. How can I monitor the heartbeat? The header of my connection includes a deviceid, and I want to update the deviceid with a Unix timestamp in Redis every heartbeat. The reason for implementing it this way is to determine whether these devices are currently online or, after killing socket.io, if these machines haven't reconnected, I could deduce they are offline by checking the expiration time along with the Unix timestamp. However, I am not able to catch the heartbeat.

**Comment by uNetworkingAB** [comment, marked as answer]
By default uWS guarantees that open/close callbacks are reliable. So you just trust them. No need to track heartbeats.

**Comment by iakuf** [comment]
I am using socket.io, which encapsulates websocket.js. However, I want to capture the last heartbeat and store it.

**Comment by ramaces** [comment]
is it normal that device goes offline on 12:20:58 and ONLINE on 12:21:48. it's contiously going online and offline ...

`compression: uWS.SHARED_COMPRESSOR,
        maxPayloadLength: 16 * 1024,
        maxLifetime: 0,
        sendPingsAutomatically: false,
        idleTimeout:45,`
